# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1456261819 28800
#      Tue Feb 23 13:10:19 2016 -0800
# Node ID 5169d5993d77e460897536976197d7b05d1df700
# Parent  030d4d6c2b58e56ef98a38ebc786a4e85ca7342a
sparse-chol.cc: Fix minor syntax error to compile with older gcc

* sparse-chol.cc (sparse_chol<SparseComplexMatrix>::sparse_chol): Remove
unnecessary "typename" keyword in explicit template declaration, fails to
compile with older versions of gcc.

diff --git a/liboctave/numeric/sparse-chol.cc b/liboctave/numeric/sparse-chol.cc
--- a/liboctave/numeric/sparse-chol.cc
+++ b/liboctave/numeric/sparse-chol.cc
@@ -556,18 +556,18 @@ chol2inv (const chol_type& r)
 
 // SparseComplexMatrix specialization (the value for the NATURAL
 // parameter in the sparse_chol<T>::sparse_chol_rep constructor is
 // different from the default).
 
 template <>
 sparse_chol<SparseComplexMatrix>::sparse_chol (const SparseComplexMatrix& a,
                                                octave_idx_type& info)
-  : rep (new typename
-         sparse_chol<SparseComplexMatrix>::sparse_chol_rep (a, info, true, false))
+  : rep (
+    new sparse_chol<SparseComplexMatrix>::sparse_chol_rep (a, info, true, false))
 { }
 
 // Instantiations we need.
 
 template class sparse_chol<SparseMatrix>;
 
 template class sparse_chol<SparseComplexMatrix>;
 
